tests/test_extract_html_content.py

```diff
@@ -0,0 +1,313 @@
+            <page id="2">
+                <name>Some Moodle Hosted Content</name>
+                <content>{(html.escape(PAGE2_CONTENT_TAG,
+                                       quote=False))}</content>
+            </page>
+        </activity>
+    """.strip()
+    return [LESSON1_CONTENT_TAGGED, PAGE2_CONTENT_TAGGED]
+
+
+def replace_ampersand(content):
+    return content.replace("&amp;", "&").replace("&", "&amp;")
+
+
+@pytest.fixture
+def mbz_path(tmp_path):
+    lesson1_content = LESSON1_XML.strip()
+    lesson1_dir = tmp_path / "activities/lesson_1"
+    lesson1_dir.mkdir(parents=True)
+    (lesson1_dir / "lesson.xml").write_text(lesson1_content)
+
+    page2_content = PAGE2_XML.strip()
+    page2_dir = tmp_path / "activities/page_2"
+    page2_dir.mkdir(parents=True)
+    (page2_dir / "page.xml").write_text(page2_content)
+
+    back_xml_content = BACKUP_XML.strip()
+    (tmp_path / "moodle_backup.xml").write_text(back_xml_content)
+
+    questions_xml_content = QUESTIONS_XML.strip()
+    (tmp_path / "questions.xml").write_text(questions_xml_content)
+
+    return tmp_path
+
+
+def test_html_files_creation(mbz_path):
+
+    # Compare file name with files in mbz.
+    html_files_list = replace_content_tags(mbz_path, mbz_path)
+    html_file_names_expected = []
+    for file in os.listdir(f"{mbz_path}"):
+        if file.endswith(".html"):
+            html_file_names_expected.append(Path(file).stem)
+
+    file_names = []
+    for file in html_files_list:
+        file_names.append(file["uuid"])
+    assert set(html_file_names_expected) == set(file_names)
+
+
+def test_html_files_content(mbz_path):
+    # compare expected html file content with files in mbz
+    html_files_list = replace_content_tags(mbz_path, mbz_path)
+    content_expected_in_files = [prettify_html(LESSON1_CONTENT1),
+                                 prettify_html(LESSON1_CONTENT2),
+                                 prettify_html(PAGE2_CONTENT)]
+
+    file_contents = []
+    for file in html_files_list:
+        file_contents.append(file["content"])
+
+    assert set(content_expected_in_files) == set(file_contents)
+
+
+def prettify_html(html):
+    soup = BeautifulSoup(html, "html.parser")
+    return soup.prettify()
+
+
+def test_xml_content_changed(mbz_path):
+    # Compare file content with files in mbz
+    html_files_list = replace_content_tags(mbz_path, mbz_path)
+    content_list_in_mbz = []
+
+    for file in glob.glob(f"{mbz_path}/activities/*/*.xml"):
+        with open(file, "r") as f:
+            content_list_in_mbz.append(f.read())
+    file_names = []
+    for file in html_files_list:
+        file_names.append(file["uuid"])
+    file_contents = []
+    for file in html_files_list:
+        file_contents.append(file["content"])
+
+    correct_content = populate_tags(file_names[0],
+                                    file_names[1], file_names[2])
+
+    content_list = set(correct_content)
+
+    assert len(content_list) == len(content_list_in_mbz)
```

Comment:
I did intend to compare the internals but removed it since I had the small differences in the xml like doctype use single quotes vs double. I do want to try something new here. Instead of checking equity I can check to see if the  
`f'<div class="os-raise-content" '  data-content-id="{ content_uuid }"></div>` is included in the xml file. To make sure that when content was extracted it was replaced with this div. Would this be a reasonable test?

Reply:
Ah -- yeah, that makes sense to me.